cMake/FreeCAD_Helpers/SetupSalomeSMESH.cmake

```diff
@@ -26,7 +26,7 @@ macro(SetupSalomeSMESH)
 
         # check which modules are available
         if(UNIX OR WIN32)
-            find_package(VTK COMPONENTS vtkCommonCore REQUIRED NO_MODULE)
+            find_package(VTK REQUIRED NO_MODULE)
```

Comment:
This is the culprit preventing builds on conda based platforms.

The file `.conda/freecad/lib/cmake/vtk-9.2/VTK-vtk-module-find-packages.cmake` has the following stanza:

    if (_vtk_module_find_package_enabled)
      set(_vtk_module_find_package_required)
      if (_vtk_module_find_package_is_required)
        set(_vtk_module_find_package_required REQUIRED)
      endif ()

      find_package(Boost
        1.85.0
        EXACT

        ${_vtk_module_find_package_quiet}
        ${_vtk_module_find_package_required}
        COMPONENTS
        OPTIONAL_COMPONENTS )
      if (NOT Boost_FOUND AND _vtk_module_find_package_fail_if_not_found)
        if (NOT ${CMAKE_FIND_PACKAGE_NAME}_FIND_QUIETLY)
          message(STATUS
            "Could not find the ${CMAKE_FIND_PACKAGE_NAME} package due to a "
            "missing dependency: Boost")
        endif ()
        set("${CMAKE_FIND_PACKAGE_NAME}_xdmf3_FOUND" 0)
        list(APPEND "${CMAKE_FIND_PACKAGE_NAME}_xdmf3_NOT_FOUND_MESSAGE"
          "Failed to find the Boost package.")
      endif ()
    endif ()

This chunk of code is evaluated when `VTK` is imported, but not when a subset of the components are imported.  It looks like the VTK component `vtkxdmfcore` is dependent upon Boost and may force this evaluation.

Reply:
That's what I get for trying to clean up the VTK warning about renaming CommonCore! I can just roll that change back, it's not important. Thanks!